File: backend/ruby_foundations/exercises/small_problems_101/easy_5/after_midnight_2.rb
MINUTES_PER_HOUR = 60
HOURS_PER_DAY = 24
MINUTES_PER_DAY = MINUTES_PER_HOUR * HOURS_PER_DAY

def after_midnight(time)
  hour = time[0..1].to_i
  minute = time[3..4].to_i
  (MINUTES_PER_HOUR * hour + minute) % MINUTES_PER_DAY
end

def before_midnight(time)
  delta_minutes = MINUTES_PER_DAY - after_midnight(time)
  delta_minutes = 0 if delta_minutes == MINUTES_PER_DAY
  delta_minutes
end

puts after_midnight('00:00') == 0
puts before_midnight('00:00') == 0
puts after_midnight('12:34') == 754
puts before_midnight('12:34') == 686
puts after_midnight('24:00') == 0
puts before_midnight('24:00') == 0

File: frontend/javascript/exercises/objects/after_midnight_2.js
var MILLISECONDS_PER_MINUTES = 60000;
var MINUTES_PER_HOUR = 60;
var HOURS_PER_DAY = 24;
var MINUTES_PER_DAY = HOURS_PER_DAY * MINUTES_PER_HOUR;

function afterMidnight(timeStr) {
  var midnight = new Date(2017, 1, 1, 0, 0);
  var timeComponents = timeStr.split(':');
  var hours = parseInt(timeComponents[0]);
  var minutes = parseInt(timeComponents[1]);
  var afterMidnight = new Date(2017, 1, 1, hours, minutes);
  var msOffset = afterMidnight.getTime() - midnight.getTime();
  return msOffset / MILLISECONDS_PER_MINUTES;
  var afterMidnight = new Date('2017/1/1')
}

function beforeMidnight(timeStr) {
  var deltaMinutes = MINUTES_PER_DAY - afterMidnight(timeStr);
  if (deltaMinutes === MINUTES_PER_DAY) {
    deltaMinutes = 0;
  }
  return deltaMinutes;
}

console.log(afterMidnight('00:00') === 0);        // 0
console.log(beforeMidnight('00:00') === 0);       // 0
console.log(afterMidnight('12:34') === 754);        // 754
console.log(beforeMidnight('12:34') === 686);       // 686